File: Python/CountValley.py
def countingValleys(steps, path):
    # Write your code here
    currHeight = 0
    isUnderCheck = False
    countValley = 0
    for step in range(steps):
        if path[step] == 'U':
            currHeight = currHeight + 1

        elif path[step] == 'D':
            currHeight = currHeight - 1

        if currHeight < 0:
            isUnderCheck = True
        else:
            if( isUnderCheck ):
                countValley = countValley + 1
                isUnderCheck = False
            else:
                isUnderCheck = False
    return countValley 

File: JS-Typescript/CountValley.js

function countingValleys(steps, path) {

    let currHeight = 0;
    let isUnderCheck = false;
    let countValley = 0;
    
    for(let i = 0; i < steps; i++){
        if(path[i] === 'U'){
            
            currHeight = currHeight + 1;
        }else if(path[i] === 'D'){
            
            currHeight = currHeight - 1;
        }
        
        if (currHeight < 0){
            isUnderCheck = true;
        }else{
            if(isUnderCheck){
                countValley = countValley + 1;
            }
            isUnderCheck = false;
        }
    }
    return countValley;
}
